src/main/java/uk/gov/hmcts/darts/common/repository/ExternalObjectDirectoryRepository.java

```diff
@@ -62,16 +62,17 @@ List<ExternalObjectDirectoryEntity> findByMediaStatusAndType(MediaEntity media,
     List<ExternalObjectDirectoryEntity> findByTranscriptionDocumentEntityAndExternalLocationType(TranscriptionDocumentEntity transcriptionDocument,
                                                                                                  ExternalLocationTypeEntity externalLocationType);
 
-    @Query(
-        """
-                SELECT eod.media.id FROM ExternalObjectDirectoryEntity eod
-                WHERE eod.media.id in :mediaIdList
```

Comment:
you mentioned in the description that you don't know why eod.media.id was used here, its because mediaIdList is a list of med_ids's , not a list of mediaEntities, eod.media refers to the entity, not the id.

Reply:
yes I noticed that, but we could have generated a list of MediaEntity by using mediaRepository.getByReference()